src/eHealth.locations/services/selected-location-factory.js

```diff
@@ -162,9 +163,12 @@ angular.module('eHealth.locations.services')
           }
         };
         level.filterByParent = function(id) {
-          level.items = original.items.filter(function (item) {
-            return item.isAll || item.parentId === id;
-          });
+          removeFilters();
```

Comment:
Maybe it would be nice to be more explicit about NOT being selected:

`removeFilters(false)`

Just a thought, though.